mlflow/store/artifact/azure_data_lake_artifact_repo.py

```diff
@@ -0,0 +1,133 @@
+import os
+import posixpath
+import re
+import urllib.parse
+
+
+from mlflow.entities import FileInfo
+from mlflow.exceptions import MlflowException
+from mlflow.store.artifact.artifact_repo import ArtifactRepository
+
+from mlflow.environment_variables import MLFLOW_ARTIFACT_UPLOAD_DOWNLOAD_TIMEOUT
+
+
+def _parse_abfss_uri(uri):
+    """
+    Parse an ABFSS URI in the format
+    "abfss://<file_system>@<account_name>.dbfs.core.windows.net/<path>",
+    returning a tuple consisting of the filesystem, account name, and path
+
+    See more details about ABFSS URIs at
+    https://learn.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-abfs-driver#uri-scheme-to-reference-data
+
+    :param uri: ABFSS URI to parse
+    :return: A tuple containing the name of the filesystem, account name, and path
+    """
+    parsed = urllib.parse.urlparse(uri)
+    if parsed.scheme != "abfss":
+        raise MlflowException("Not an ABFSS URI: %s" % uri)
+
+    match = re.match(r"([^@]+)@([^.]+)\.dfs\.core\.windows\.net", parsed.netloc)
+
+    if match is None:
+        raise MlflowException(
+            "ABFSS URI must be of the form abfss://<filesystem>@<account>.dfs.core.windows.net"
+        )
+    filesystem = match.group(1)
+    account_name = match.group(2)
+    path = parsed.path
+    if path.startswith("/"):
+        path = path[1:]
+    return filesystem, account_name, path
+
+
+def _get_data_lake_client(account_url, credential):
+    from azure.storage.filedatalake import DataLakeServiceClient
+
+    return DataLakeServiceClient(account_url, credential)
+
+
+class AzureDataLakeArtifactRepository(ArtifactRepository):
+    """
+    Stores artifacts on Azure Data Lake Storage Gen2.
+
+    This repository is used with URIs of the form
+    ``abfs[s]://file_system@account_name.dfs.core.windows.net/<path>/<path>``.
+
+    :param credential: Azure credential (see options in https://learn.microsoft.com/en-us/python/api/azure-core/azure.core.credentials?view=azure-python)
+                       to use to authenticate to storage
+    """
+
+    def __init__(self, artifact_uri, credential):
+        super().__init__(artifact_uri)
+        _DEFAULT_TIMEOUT = 600  # 10 minutes
+        self.write_timeout = MLFLOW_ARTIFACT_UPLOAD_DOWNLOAD_TIMEOUT.get() or _DEFAULT_TIMEOUT
```

Comment:
Is this attribute `self.write_timeout` used somewhere?

Reply:
It's not, thanks for the catch :)